foundry/pixi/containers/canvasLayers/sightLayer.d.ts

```diff
@@ -1 +1,343 @@
+   * Return a vision container back to the pool, recycling it for future use.
+   * @param c - The container to recycle
+   */
+  protected _recycleVisionContainer(c: PIXI.Container): void;
+
+  /**
+   * Update the display of the sight layer.
+   * Organize sources into rendering queues and draw lighting containers for each source
+   *
+   * @param forceUpdateFog - Always update the Fog exploration progress for this update
+   *                         (default: `false`)
+   * @param noUpdateFog    - Never update the Fog exploration progress for this update
+   *                         (default: `false`)
+   */
+  refresh({ forceUpdateFog, noUpdateFog }?: { forceUpdateFog?: boolean; noUpdateFog?: boolean }): void;
+
+  /* -------------------------------------------- */
+
+  /**
+   * Restrict the visibility of certain canvas assets (like Tokens or DoorControls) based on the visibility polygon
+   * These assets should only be displayed if they are visible given the current player's field of view
+   */
+  restrictVisibility(): void;
+
+  /**
+   * Once a new Fog of War location is explored, composite the explored container with the current staging sprite
+   * Save that staging Sprite as the rendered fog exploration and swap it out for a fresh staging texture
+   * Do all this asynchronously, so it doesn't block token movement animation since this takes some extra time
+   */
+  protected commitFog(): Promise<void>;
+
+  /**
+   * Load existing fog of war data from local storage and populate the initial exploration sprite
+   */
+  loadFog(): Promise<PIXI.Texture>;
+
+  /**
+   * Dispatch a request to reset the fog of war exploration status for all users within this Scene.
+   * Once the server has deleted existing FogExploration documents, the _onResetFog handler will re-draw the canvas.
+   */
+  resetFog(): Promise<Canvas | undefined>;
+
+  /**
+   * Save Fog of War exploration data to a base64 string to the FogExploration document in the database.
+   * Assumes that the fog exploration has already been rendered as fog.rendered.texture.
+   */
+  protected saveFog(): Promise<SocketInterface.Responses.ModifyEmbeddedDocument | undefined>;
+
+  /**
+   * Update the fog layer when a player token reaches a board position which was not previously explored
+   * @param source - The vision source for which the fog layer should update
+   * @param force  - Force fog to be updated even if the location is already explored
+   *                 (default: `false`)
+   */
+  updateFog(source: PointSource, force?: boolean): void;
+
+  /**
+   * Choose an adaptive fog rendering resolution which downscales the saved fog textures for larger dimension Scenes
+   */
+  protected _configureFogResolution(): number;
+
+  /**
+   * Trigger a server-side update (or creation) of fog exploration status for a certain Scene
+   */
+  protected _createOrUpdateFogExploration(
+    fogData: this['fogData']
+  ): Promise<SocketInterface.Responses.ModifyEmbeddedDocument>;
```

Comment:
Are you sure `ModifyEmbeddedDocument` is correct? The request is called `"modifyDocument"`. I think we just didn't completely type the `SocketInterface` yet. It's probably missing a lot of types. I would just type this as `Promise<any>` for now and add a TODO to update it when `SocketInterface` is being handled.